core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -267,11 +279,35 @@ public AtomicReference<Long> getWorkerId() {
   }
 
   @Override
-  public UfsStatus[] listStatus(String path, ListOptions options) throws IOException {
-    UfsStatus[] statuses = mListStatusCache.getIfPresent(path);
+  public UfsStatus[] listStatus(String path, ListStatusPOptions options) throws IOException {
+    UfsStatus[] statuses;
+    ListOptions listOptions = ListOptions.defaults().setRecursive(
+        options.hasRecursive() ? options.getRecursive() : false);
+
+    long syncIntervalMs = options.hasCommonOptions()
+        ? (options.getCommonOptions().hasSyncIntervalMs()
+        ? options.getCommonOptions().getSyncIntervalMs() : -1) :
+        -1;
+
+    ListStatusResult result = mListStatusCache.getIfPresent(path);
+    if (syncIntervalMs >= 0 && result != null) {
+      // Check if the metadata is still valid.
+      if (System.nanoTime() - result.mTimeStamp > syncIntervalMs * Constants.MS_NANO) {
+        // The metadata is expired. Remove it from in-memory cache.
+        mListStatusCache.invalidate(path);
+        result = null;
+      }
+    }
+
+    if (result != null) {
+      statuses = result.mUfsStatuses;
+    } else {
+      statuses = null;
+    }
```

Comment:
I'd recommend to avoid using a generic name like `result` `statuses` for variables and reusing them for multiple purposes. Also I'd recommend to mark variables `final` to make sure they are not changed after initialization, and what they contains is exactly as described by their name.

```suggestion
    final UfsStatus[] cachedStatuses;
    final ListStatusResult resultFromCache = mListStatusCache.getIfPresent(path);
    if (resultFromCache == null) {
      cachedStatuses = null;
    } else {
      // Metadata is cached, check if the metadata has expired
      if (syncIntervalMs >= 0 
          && System.nanoTime() - resultFromCache.mTimeStamp > syncIntervalMs * Constants.MS_NANO) {
        // The metadata has expired. Remove it from in-memory cache.
        mListStatusCache.invalidate(path);
        cachedStatuses = null;
      } else {
        // Still valid, use cached statuses
        cachedStatuses = resultFromCache.mUfsStatuses;
      }
    }
```

Reply:
OK